kie-server-parent/kie-server-services/kie-server-services-jbpm-queries/src/test/java/org/kie/server/services/jbpm/queries/ProcessInstanceQueryServiceBaseTest.java

```diff
@@ -0,0 +1,109 @@
+ * limitations under the License.
+*/
+
+package org.kie.server.services.jbpm.queries;
+
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.*;
+import static org.mockito.Mockito.*;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.jbpm.services.api.model.ProcessInstanceWithVarsDesc;
+import org.jbpm.services.api.query.QueryService;
+import org.jbpm.services.api.query.model.QueryParam;
+import org.junit.Test;
+import org.kie.server.api.model.KieServerConfig;
+import org.kie.server.api.model.instance.ProcessInstance;
+import org.kie.server.api.model.instance.ProcessInstanceList;
+import org.kie.server.jbpm.queries.api.model.definition.ProcessInstanceQueryFilterSpec;
+import org.kie.server.services.api.KieServerRegistry;
+import org.kie.server.services.jbpm.queries.ProcessInstanceQueryServiceBase;
+import org.kie.server.services.jbpm.queries.util.QueryStrategy;
+import org.mockito.Mockito;
+
+public class ProcessInstanceQueryServiceBaseTest {
+	
+	@Test
+	public void testGetProcessInstancesTasksWithFilters() {
+		QueryService queryServiceMock = Mockito.mock(QueryService.class);
+		KieServerRegistry contextMock = Mockito.mock(KieServerRegistry.class);
+		
+		
+		//Registry mock needs to return extra classes registered by the extension.
+		Set<Class<?>> extraClasses = new HashSet<>();
+		extraClasses.add(ProcessInstanceQueryFilterSpec.class);
+		when(contextMock.getExtraClasses()).thenReturn(extraClasses);
+		
+		KieServerConfig configMock = Mockito.mock(KieServerConfig.class);
+		when(contextMock.getConfig()).thenReturn(configMock);
+		
+		QueryStrategy processInstanceQueriesStrategyMock = Mockito.mock(QueryStrategy.class); 
+		
+		ProcessInstanceQueryServiceBase base = new ProcessInstanceQueryServiceBase(queryServiceMock, contextMock, processInstanceQueriesStrategyMock);
+		
+		//Verify that extra classes are registered on the context.
+		verify(contextMock, times(1)).addExtraClasses(extraClasses);
+		
+		Integer page = new Integer(0);
+		Integer pageSize = new Integer(10);
+		String payload = getPayload();
+		String marshallingType = "JAXB";
+		
+		Collection<ProcessInstanceWithVarsDesc> processInstanceCollection = getProcessInstances();
+		
+		when(queryServiceMock.query(any(), any(), any(), any(QueryParam.class))).thenReturn(processInstanceCollection);
+		
+		ProcessInstanceList processInstances = base.getProcessInstancesWithFilters(page, pageSize, payload, marshallingType);
+		
+		//TODO: Implement the same logic with JUnit AssertThat
+		assertEquals(1, processInstances.getItems().size());
+		
+		ProcessInstance pi1 = processInstances.getItems().stream().findFirst().get();
+		
+		assertEquals("test-process", pi1.getProcessName());
+		assertEquals(42L, pi1.getId().longValue());
+		assertEquals("mswiderski", pi1.getInitiator());
+	}
+	
+	private String getPayload() {
```

Comment:
Is there any better way to build/create the query and put it into payload? builder for example?